atomate/qchem/drones.py

```diff
@@ -180,36 +183,47 @@ def generate_doc(self, dir_name, qcinput_files, qcoutput_files, multirun):
                     d["output"]["final_energy"] = d["calcs_reversed"][1][
                         "final_energy"]
 
+            if d["output"]["job_type"] == "sp":
+                d["output"]["final_energy"] = d_calc_final["SCF"][0][-1][0]
+
             if "special_run_type" in d:
                 if d["special_run_type"] == "frequency_flattener":
-                    d["num_frequencies_flattened"] = (
-                        len(qcinput_files) / 2) - 1
-
-            total_cputime = 0.0
-            total_walltime = 0.0
-            nan_found = False
-            for calc in d["calcs_reversed"]:
-                if calc["walltime"] != "nan":
-                    total_walltime += calc["walltime"]
+                    d["num_frequencies_flattened"] = int((len(qcinput_files) / 2) - 1)
+
+            if d_calc_final["completion"]:
+                total_cputime = 0.0
+                total_walltime = 0.0
+                nan_found = False
+                for calc in d["calcs_reversed"]:
+                    if calc["walltime"] != "nan":
+                        total_walltime += calc["walltime"]
+                    else:
+                        nan_found = True
+                    if calc["cputime"] != "nan":
+                        total_cputime += calc["cputime"]
+                    else:
+                        nan_found = True
+                if nan_found:
+                    d["walltime"] = "nan"
+                    d["cputime"] = "nan"
                 else:
-                    nan_found = True
-                if calc["cputime"] != "nan":
-                    total_cputime += calc["cputime"]
-                else:
-                    nan_found = True
-            if nan_found:
-                d["walltime"] = "nan"
-                d["cputime"] = "nan"
+                    d["walltime"] = total_walltime
+                    d["cputime"] = total_cputime
             else:
-                d["walltime"] = total_walltime
-                d["cputime"] = total_cputime
+                d["walltime"] = "NA"
```

Comment:
why is it sometimes "NA" and sometimes "nan"? Very confused what all these things mean.

I would just either have a number if you know the number or None if you don't know the number for whatever reason

Reply:
Fixed